src/lib.rs

```diff
@@ -236,18 +239,49 @@ mod test {
     #[test]
     fn test_read_array() {
         let a = &[42];
-        assert!(read_array::<Foo>(a) == &[Foo { a: 42 }]);
+        assert_eq!(read_array::<Foo>(a), &[Foo { a: 42 }]);
         let a = &[42, 43, 44, 45];
-        assert!(read_array::<Foo>(a) == &[Foo { a: 42 }, Foo { a: 43 }, Foo { a: 44 }, Foo { a: 45 }]);
-
-        let a = &[42, 0, 0, 0, 0x03, 0xff, 0x62, 0xa2, 0x5b, 0x42, 0x00, 0xf0, -2i8 as u8];
-        assert!(read_array::<Bar>(a) == &[Bar { a: 42, b: 0xf000425b_a262ff03, c: -2 }]);
-        let a = &[42, 0, 0, 0, 0x03, 0xff, 0x62, 0xa2, 0x5b, 0x42, 0x00, 0xf0, -2i8 as u8,
-                  43, 0, 0, 0, 0x03, 0xff, 0x62, 0xa2, 0x5b, 0x42, 0x00, 0xf0, -2i8 as u8,
-                  44, 0, 0, 0, 0x03, 0xff, 0x62, 0xa2, 0x5b, 0x42, 0x00, 0xf0, -2i8 as u8];
-        assert!(read_array::<Bar>(a) == &[Bar { a: 42, b: 0xf000425b_a262ff03, c: -2 },
-                                          Bar { a: 43, b: 0xf000425b_a262ff03, c: -2 },
-                                          Bar { a: 44, b: 0xf000425b_a262ff03, c: -2 }]);
```

Comment:
I think the formatting of these arrays is useful, so better to use `#[rustfmt::skip]`